jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1983,4 +1981,56 @@ private IllegalCharacterException(State state,HttpTokens.Token token,ByteBuffer
                 LOG.debug(String.format("Illegal character %s in state=%s for buffer %s",token,state,BufferUtil.toDetailString(buffer)));
         }
     }
+
+    public interface ComplianceViolation
```

Comment:
Why the interface?    **If** it is needed, it should have javadoc to explain why.

Reply:
Also why is it in HttpParser?

Reply:
Oh I now see why the interface... so you can have a single listener for multiple types of violation!